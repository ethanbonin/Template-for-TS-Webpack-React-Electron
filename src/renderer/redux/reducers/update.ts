import {
    UpdateInitState,
    UpdateActions,
    UPDATE_DOWNLOADED,
    UPDATE_AVAILABLE,
} from '@/renderer/types/update';

const initialState: UpdateInitState = {
    updateAvailable: false,
    updateDownloaded: false,
};

export default function update(
    state: UpdateInitState = initialState,
    action: UpdateActions,
) {
    switch (action.type) {
        case UPDATE_AVAILABLE:
            return { ...state, updateAvailable: !state.updateAvailable };
        case UPDATE_DOWNLOADED:
            return { ...state, updateDownloaded: !state.updateDownloaded };
        default:
            return state;
    }
}
