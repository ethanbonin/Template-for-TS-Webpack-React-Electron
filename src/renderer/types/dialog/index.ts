export enum DialogBoxType {
    none = 'none',
    info = 'info',
    error = 'error',
    question = 'question',
    warning = 'warning',
}

export interface DialogProperties {
    type: DialogBoxType;
    buttons?: string[];
    defaultId?: number;
    title?: string;
    message: string;
    detail?: string;
    checkboxLabel?: string;
    checkboxChecked?: boolean;
}

export interface DialogInitState {
    dialogProperties: DialogProperties;
}

export const SHOW_DIALOG = 'SHOW_DIALOG';

export interface ShowDialog {
    type: typeof SHOW_DIALOG;
    dialogProperties: DialogProperties;
}

export type DialogActions = ShowDialog;
