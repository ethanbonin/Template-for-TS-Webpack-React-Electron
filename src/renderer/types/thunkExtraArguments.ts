export interface ThunkExtraArguments {
    extraAddedArgumentExample: string;
}
