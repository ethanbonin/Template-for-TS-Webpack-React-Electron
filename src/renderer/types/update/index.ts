export interface UpdateInitState {
    updateAvailable: boolean;
    updateDownloaded: boolean;
}

export const UPDATE_AVAILABLE = 'UPDATE_AVAILABLE';
export const UPDATE_DOWNLOADED = 'UPDATE_DOWNLOADED';

export interface SetUpdateAvailable {
    type: typeof UPDATE_AVAILABLE;
}

export interface SetUpdateDownloaded {
    type: typeof UPDATE_DOWNLOADED;
}

export type UpdateActions = SetUpdateAvailable | SetUpdateDownloaded;
