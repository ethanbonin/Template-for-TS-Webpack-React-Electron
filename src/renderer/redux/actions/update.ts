import { push } from 'connected-react-router';
import { AppThunk } from '@/renderer/types/redux';
import { UPDATE_AVAILABLE, UPDATE_DOWNLOADED } from '@renderer/types/update/';
import { routes } from '@/renderer/constants/routes';

export const toggleUpdateDownloaded: AppThunk = () => async (dispatch) => {
    return dispatch({
        type: UPDATE_DOWNLOADED,
    });
};

export const toggleUpdateAvailable: AppThunk = () => async (dispatch) => {
    dispatch(toggleUpdateDownloaded());
    return dispatch({
        type: UPDATE_AVAILABLE,
    });
};

export const goToNextPage: AppThunk = () => async (dispatch, getState, extraArgument) => {
    const state = getState();
    const { update } = state;

    console.log('State Example: { update }', update);
    console.log('Extra Argument Example:', extraArgument);

    dispatch(toggleUpdateAvailable());
    dispatch(push(routes.secondPage));
};
