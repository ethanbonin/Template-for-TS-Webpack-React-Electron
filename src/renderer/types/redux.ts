import { ThunkAction, ThunkDispatch as TD } from 'redux-thunk';
import { AppState } from '@/renderer/store';
import { AllActions } from '@renderer/redux/actions/';
import { ThunkExtraArguments } from '@/renderer/types/thunkExtraArguments';
import { ActionCreator } from 'redux';

export type AppThunk = ActionCreator<
    ThunkAction<
        Promise<void> | Promise<AllActions>,
        AppState | undefined,
        ThunkExtraArguments | undefined,
        AllActions
    >
>;
export type ThunkDispatch = TD<AppState, ThunkExtraArguments, AllActions>;
