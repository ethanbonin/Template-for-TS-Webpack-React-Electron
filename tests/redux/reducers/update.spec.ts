import update from '@/renderer/redux/reducers/update';
import { UPDATE_AVAILABLE, UPDATE_DOWNLOADED } from '@/renderer/types/update';

describe('update reducer', () => {
    it('should handle initial state', () => {
        const state = update(undefined, { type: UPDATE_AVAILABLE });
        const updatedState = update(state, { type: UPDATE_AVAILABLE }); // Reset it
        expect(updatedState).toMatchSnapshot();
    });

    it('should handle UPDATE_AVAILABLE', () => {
        const state = update(undefined, { type: UPDATE_AVAILABLE });
        expect(state.updateAvailable).toBe(true);

        expect(state.updateDownloaded).toBe(false);

        const updateState = update(state, { type: UPDATE_AVAILABLE });
        expect(updateState.updateAvailable).toBe(false);
    });

    it('should handle UPDATE_DOWNLOADED', () => {
        const state = update(undefined, { type: UPDATE_DOWNLOADED });
        expect(state.updateDownloaded).toBe(true);

        expect(state.updateAvailable).toBe(false);

        const updateState = update(state, { type: UPDATE_DOWNLOADED });
        expect(updateState.updateDownloaded).toBe(false);
    });
});
