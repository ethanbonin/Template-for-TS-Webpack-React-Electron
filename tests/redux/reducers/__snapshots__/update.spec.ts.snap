// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update reducer should handle initial state 1`] = `
Object {
  "updateAvailable": false,
  "updateDownloaded": false,
}
`;
